ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/BufferedTransactionsReader.scala

```diff
@@ -155,20 +155,20 @@ private[platform] object BufferedTransactionsReader {
       outputStreamBufferSize: Int,
       bufferSizeCounter: Counter,
   )(implicit executionContext: ExecutionContext): Source[(Offset, API_RESPONSE), NotUsed] = {
-    def filterBuffered(
+    def bufferedSource(
         slice: Vector[(Offset, TransactionLogUpdate)]
-    ): Future[Iterator[(Offset, API_RESPONSE)]] =
-      Future
-        .traverse(
-          slice.iterator
-            .collect { case (offset, tx: TxUpdate) =>
-              toApiTx(tx, filter, verbose).map(offset -> _)
-            }
-        )(identity)
-        .map(_.collect { case (offset, Some(tx)) =>
+    ): Source[(Offset, API_RESPONSE), NotUsed] =
+      Source
+        .fromIterator(() => slice.iterator)
+        .collect { case (offset, tx: TxUpdate) =>
+          toApiTx(tx, filter, verbose).map(offset -> _)
+        }
+        .mapAsync(32)(identity)
```

Comment:
Wat.

Reply:
Are you worried about the construction or the parallelism level?

Reply:
I am confused about what this is supposed to do.

Does it flatten the futures or something?

Reply:
How will this mapAsynch behave on a machine with a meager number of CPUs? Not every deployment will use a massive parallelism.

Reply:
`mapAsync` does not use the `ActorSystem'`s thread allocation for running the tasks unless we specifically instruct the Futures within to run on a direct execution context, which we don't. The actual parallelism limit in that operator is given by the execution context thread size cap on which the Futures are executed.

Reply:
Added a comment clarifying in code the current status